nx/lib/nx/binary_backend.ex

```diff
@@ -1634,10 +1657,20 @@ defmodule Nx.BinaryBackend do
             end
           end
 
-        IO.iodata_to_binary(Enum.sort(data, comparator))
+        sorted =
```

Comment:
I'm not sure if the `match_types` call on line 1654 is needed. Perhaps we should remove it or refactor the `for`-loop somehow to use it correctly

Reply:
It isn't, you can remove it.